src/io/flutter/view/FlutterView.java

```diff
@@ -254,27 +256,33 @@ protected void perform(AnActionEvent event) {
 }
 
 class TogglePlatformAction extends AbstractFlutterAction {
-  private boolean isAndroid = false;
-
   TogglePlatformAction(@NotNull FlutterView view) {
     super(view, FlutterBundle.message("flutter.view.togglePlatform.text"), FlutterBundle.message("flutter.view.togglePlatform.description"),
           AllIcons.RunConfigurations.Application);
   }
 
   @Override
   public void actionPerformed(AnActionEvent event) {
-    isAndroid = !isAndroid;
-
     // ext.flutter.platformOverride, value: iOS, android
-    final Map<String, Object> params = new HashMap<>();
-    params.put("value", isAndroid ? "android" : "iOS");
-    view.getFlutterApp().callServiceExtension("ext.flutter.platformOverride", params);
+    final CompletableFuture<JsonObject> result = view.getFlutterApp().callServiceExtension(
+      "ext.flutter.platformOverride", new HashMap<>());
+    result.thenAccept(obj -> {
+      if (obj == null) {
+        return;
+      }
 
-    final ConsoleView console = view.getFlutterApp().getConsole();
-    if (console != null) {
-      console.print(FlutterBundle.message("flutter.view.togglePlatform.output", isAndroid ? "Android" : "iOS"),
-                    ConsoleViewContentType.SYSTEM_OUTPUT);
-    }
+      final boolean switchToAndroid = !"android".equals(obj.get("value").getAsString());
+
+      final Map<String, Object> params = new HashMap<>();
+      params.put("value", switchToAndroid ? "android" : "iOS");
+      view.getFlutterApp().callServiceExtension("ext.flutter.platformOverride", params);
+
+      final ConsoleView console = view.getFlutterApp().getConsole();
+      if (console != null) {
+        console.print(FlutterBundle.message("flutter.view.togglePlatform.output", switchToAndroid ? "Android" : "iOS"),
```

Comment:
The console message is: `Switched platform OS to {0}`

But, technically, we don't know that it worked since we didn't wait for the response to the second call. Maybe call thenAccept() again?